pkg/distributor/ha_tracker.go

```diff
@@ -220,16 +235,24 @@ func (c *haTracker) checkReplica(ctx context.Context, userID, cluster, replica s
 	}
 	key := fmt.Sprintf("%s/%s", userID, cluster)
 	now := mtime.Now()
+
 	c.electedLock.RLock()
 	entry, ok := c.elected[key]
+	clusters := c.clusters[userID]
 	c.electedLock.RUnlock()
+
 	if ok && now.Sub(timestamp.Time(entry.ReceivedAt)) < c.cfg.UpdateTimeout+c.updateTimeoutJitter {
 		if entry.Replica != replica {
 			return replicasNotMatchError(replica, entry.Replica)
 		}
 		return nil
 	}
 
+	// We don't know about this cluster yet. If we have reached the limit for number of clusters, we error out now.
```

Comment:
To my understanding of this function, we may already know about the cluster if `ok` is true, but code could reach t his point anyway. Am I wrong?

Reply:
Good point! I will try to add a test for this.

Reply:
Should be fixed, and added a test too.